packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Groups/common/GroupForm/AllowList.tsx

```diff
@@ -0,0 +1,272 @@
+    .namespace;
+
+  const tableState = useCWTableState({
+    columns: tableColumns(isStakedCommunity),
+    initialSortDirection: APIOrderDirection.Desc,
+  });
+
+  const membersPerPage = 10;
+  const { fetchNextMembersPage, groups, isLoadingMembers, members } =
+    useMemberData({
+      tableState,
+      searchFilters,
+      memberships,
+      membersPerPage,
+    });
+
+  const handleChange = async (_e, page: number) => {
+    setCurrentPage(page);
+    await fetchNextMembersPage({ pageParam: page });
+  };
+  let totalPages = members?.pages?.[0].totalPages ?? 0;
+
+  if (searchFilters.groupFilter === 'allowlisted') {
+    totalPages = Math.ceil(allowedAddresses.length / membersPerPage);
+  } else if (searchFilters.groupFilter === 'not-allowlisted') {
+    totalPages =
+      totalPages - Math.ceil(allowedAddresses.length / membersPerPage);
+  }
+
+  const formattedMembers: Member[] = useMemo(() => {
+    if (!members?.pages?.length) {
+      return [];
+    }
+
+    let memberResults =
+      members?.pages?.find((p) => p.page === currentPage)?.results ?? [];
+
+    if (
+      searchFilters.groupFilter === 'allowlisted' ||
+      searchFilters.groupFilter === 'not-allowlisted'
+    ) {
+      memberResults = members.pages
+        .flatMap((p) => p.results)
+        .filter((r) => {
+          if (allowedAddresses.includes(r.addresses[0].address)) {
+            return searchFilters.groupFilter === 'allowlisted';
+          }
+          return searchFilters.groupFilter === 'not-allowlisted';
+        });
+
+      memberResults = memberResults.slice(
+        (currentPage - 1) * membersPerPage,
+        currentPage * membersPerPage,
+      );
+
+      const uniqueMembers = new Map();
+      memberResults.forEach((m) => {
+        uniqueMembers.set(m.id, m);
+      });
+
+      memberResults = Array.from(uniqueMembers.values());
+    }
+
+    return memberResults.map((p) => ({
+      id: p.id,
+      avatarUrl: p.avatar_url,
+      name: p.profile_name || 'Anonymous',
+      role: p.roles[0],
+      groups: (p.group_ids || [])
+        .map(
+          (groupId) =>
+            (groups || []).find((group) => group.id === groupId)?.name,
+        )
+        .filter(Boolean)
+        .sort((a, b) => a.localeCompare(b)),
+      stakeBalance: p.addresses[0].stake_balance,
+      address: p.addresses[0].address,
+    })) as Member[];
+    // we disable the exhaustive-deps because we don't want to refresh on changed allowedAddresses because it will
+    // update the displayed list while the boxes are being checked which is a bit jarring from a UI perspective
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchFilters.groupFilter, members?.pages, currentPage, groups]);
+
+  useEffect(() => {
+    // Invalidate group memberships cache
+    void queryClient.cancelQueries([ApiEndpoints.FETCH_GROUPS]);
+    void queryClient.refetchQueries([ApiEndpoints.FETCH_GROUPS]);
+  }, []);
```

Comment:
I don't like that this is here. More context?

Reply:
I am actually also not sure, but we also use it on the memberships page, so I figured we would need it here since we use a bunch of the same queries.

Do you remember why this was needed @mzparacha?

Reply:
anyways I removed them for the allowlists because they don't seem to be necessary.